src/engine/UI/DisplayLayout.cpp

```diff
@@ -7,8 +7,8 @@ using namespace admirals::UI;
 DisplayLayout::DisplayLayout()
     : m_font(Texture::loadFromPath("assets/font.png")) {}
 
-void DisplayLayout::AddElement(std::shared_ptr<Element> element) {
-    m_elements.push_back(std::move(element));
+void DisplayLayout::AddElement(const std::shared_ptr<Element> &element) {
```

Comment:
Shared pointers should in the general case be passed as straight copies, not by reference. This follows the semantics of a shared pointer that it increments a reference counter in every context it is "shared" in. By passing a reference to it, the underlying reference counter is not incremented and this context will not count. Would essentially work as a `std::weak_ptr`.